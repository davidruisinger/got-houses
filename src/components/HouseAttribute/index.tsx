import React, { FunctionComponent, ReactChild } from 'react'
import posed from 'react-pose'
import styled from '../_utils/styledComponents'
import FlexBox from '../FlexBox'
import Container from '../Container'

const animationProps = {
  enter: {
    x: 0,
    opacity: 1,
    delay: ({ i }: { i: number }) => i * 100,
    beforeChildren: true,
  },
  exit: { x: 50, opacity: 0 },
}

const AnimatedWrapper = posed(FlexBox)(animationProps)

const Icon = styled.div`
  font-size: 2.8em;
  padding: 0.2em;

  color: ${props => props.theme.text_default};
  background-color: ${props => props.theme.canvas_ground};
`

interface Props {
  icon: ReactChild
  data: string
  i?: number
}

const HouseAttribute: FunctionComponent<Props> = ({ icon, data, i }) => (
  <AnimatedWrapper i={i} justifyContent="space-between" alignItems="stretch">
    <Icon>{icon}</Icon>
    <Container>{data}</Container>
  </AnimatedWrapper>
)

HouseAttribute.defaultProps = {
  i: 1,
}

export default HouseAttribute
