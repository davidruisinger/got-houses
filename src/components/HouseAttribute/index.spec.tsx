import 'jest-styled-components'
import React from 'react'
import { shallow, mount } from 'enzyme'
import HouseAttribute from './index'
import { GiCrossedSwords } from 'react-icons/gi'

describe('<HouseAttribute />', () => {
  const data = 'Test data string'
  const shallowedWraper = shallow(
    <HouseAttribute icon={<GiCrossedSwords />} data={data} />
  )
  const mountedWrapper = mount(
    <HouseAttribute icon={<GiCrossedSwords />} data={data} />
  )

  it('matches the snapshot', () => {
    expect(shallowedWraper).toMatchSnapshot()
  })

  it('renders the icon', () => {
    expect(mountedWrapper.find(GiCrossedSwords).length).toBe(1)
  })

  it('renders the data', () => {
    expect(
      mountedWrapper
        .find('p')
        .first()
        .text()
    ).toEqual(data)
  })
})
