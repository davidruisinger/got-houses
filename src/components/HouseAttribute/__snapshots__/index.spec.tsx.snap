// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HouseAttribute /> matches the snapshot 1`] = `
<ForwardRef
  alignItems="stretch"
  i={1}
  justifyContent="space-between"
>
  <styled.div>
    <GiCrossedSwords />
  </styled.div>
  <styled.div
    textAlign="left"
  >
    <p>
      Test data string
    </p>
  </styled.div>
</ForwardRef>
`;
